File: internal/dto/art.go
package dto

import "github.com/google/uuid"

type (
	ArtResponse struct {
		ID           uuid.UUID    `json:"id"`
		Author       UserResponse `json:"author"`
		Corner       string       `json:"corner"`
		ArtType      string       `json:"art_type"`
		Title        string       `json:"title"`
		Description  string       `json:"description"`
		ImageURL     string       `json:"image_url"`
		ThumbnailURL string       `json:"thumbnail_url"`
		GalleryID    *uuid.UUID   `json:"gallery_id,omitempty"`
		Tags         []string     `json:"tags"`
		LikeCount    int          `json:"like_count"`
		CommentCount int          `json:"comment_count"`
		ViewCount    int          `json:"view_count"`
		UserLiked    bool         `json:"user_liked"`
		IsSpoiler    bool         `json:"is_spoiler"`
		CreatedAt    string       `json:"created_at"`
		UpdatedAt    *string      `json:"updated_at,omitempty"`
	}

	ArtDetailResponse struct {
		ArtResponse
		Comments      []ArtCommentResponse `json:"comments"`
		LikedBy       []UserResponse       `json:"liked_by"`
		ViewerBlocked bool                 `json:"viewer_blocked"`
	}

	CreateArtRequest struct {
		Title       string     `json:"title"`
		Description string     `json:"description"`
		Corner      string     `json:"corner"`
		ArtType     string     `json:"art_type"`
		Tags        []string   `json:"tags"`
		IsSpoiler   bool       `json:"is_spoiler"`
		GalleryID   *uuid.UUID `json:"gallery_id,omitempty"`
	}

	UpdateArtRequest struct {
		Title       string   `json:"title"`
		Description string   `json:"description"`
		Tags        []string `json:"tags"`
		IsSpoiler   bool     `json:"is_spoiler"`
	}

	ArtCommentResponse struct {
		ID        uuid.UUID            `json:"id"`
		ParentID  *uuid.UUID           `json:"parent_id,omitempty"`
		Author    UserResponse         `json:"author"`
		Body      string               `json:"body"`
		Media     []PostMediaResponse  `json:"media"`
		LikeCount int                  `json:"like_count"`
		UserLiked bool                 `json:"user_liked"`
		Replies   []ArtCommentResponse `json:"replies,omitempty"`
		CreatedAt string               `json:"created_at"`
		UpdatedAt *string              `json:"updated_at,omitempty"`
	}

	ArtListResponse struct {
		Art    []ArtResponse `json:"art"`
		Total  int           `json:"total"`
		Limit  int           `json:"limit"`
		Offset int           `json:"offset"`
	}

	TagCountResponse struct {
		Tag   string `json:"tag"`
		Count int    `json:"count"`
	}

	GalleryResponse struct {
		ID                uuid.UUID         `json:"id"`
		Author            UserResponse      `json:"author"`
		Name              string            `json:"name"`
		Description       string            `json:"description"`
		CoverImageURL     string            `json:"cover_image_url"`
		CoverThumbnailURL string            `json:"cover_thumbnail_url"`
		PreviewImages     []PreviewImageDTO `json:"preview_images,omitempty"`
		ArtCount          int               `json:"art_count"`
		CreatedAt         string            `json:"created_at"`
		UpdatedAt         *string           `json:"updated_at,omitempty"`
	}

	PreviewImageDTO struct {
		Thumbnail string `json:"thumbnail_url"`
		Full      string `json:"full_url"`
	}

	CreateGalleryRequest struct {
		Name        string `json:"name"`
		Description string `json:"description"`
	}

	UpdateGalleryRequest struct {
		Name        string `json:"name"`
		Description string `json:"description"`
	}
)

File: frontend/src/types/api/art.ts
import type { CommentBase, PaginationFields, User } from "./common";

export interface Art {
    id: string;
    author: User;
    corner: string;
    art_type: string;
    title: string;
    description: string;
    image_url: string;
    thumbnail_url: string;
    gallery_id?: string;
    tags: string[];
    like_count: number;
    comment_count: number;
    view_count: number;
    user_liked: boolean;
    is_spoiler: boolean;
    created_at: string;
    updated_at?: string;
}

export interface ArtDetail extends Art {
    comments: ArtComment[];
    liked_by: User[];
    viewer_blocked: boolean;
}

export interface ArtComment extends CommentBase {
    replies?: ArtComment[];
}

export interface ArtListResponse extends PaginationFields {
    art: Art[];
}

export interface TagCount {
    tag: string;
    count: number;
}

export interface Gallery {
    id: string;
    author: User;
    name: string;
    description: string;
    cover_image_url: string;
    cover_thumbnail_url: string;
    preview_images?: { thumbnail_url: string; full_url: string }[];
    art_count: number;
    created_at: string;
    updated_at?: string;
}

export interface GalleryDetailResponse extends PaginationFields {
    gallery: Gallery;
    art: Art[];
}
